internal/data/trace.go

```diff
@@ -51,21 +51,21 @@ func (td TraceData) ResourceSpans() []*ResourceSpans {
 type ResourceSpans struct {
 	// The resource for the spans in this message.
 	// If this field is not set then no resource info is known.
-	resource *Resource
+	resource Resource
 
 	// A list of Spans that originate from a resource.
 	spans []*Span
 }
 
-func NewResourceSpans(resource *Resource, spans []*Span) *ResourceSpans {
+func NewResourceSpans(resource Resource, spans []*Span) *ResourceSpans {
 	return &ResourceSpans{resource, spans}
 }
 
-func (m *ResourceSpans) Resource() *Resource {
+func (m *ResourceSpans) Resource() Resource {
```

Comment:
This will return a copy, so any modifications you apply on a returned resource won't have any effect in the original `ResourceSpans` object. Is that intended?

Reply:
It returns a copy with pointers to actual data. Changes are only done in the pointed data so will be visible on all copies. The link that you have in this comment shows code that no longer exist, it is changed to use pimpl pattern in this PR.

Reply:
Makes sense, thanks.